core/src/game_state.rs

```diff
@@ -1139,7 +1139,10 @@ impl PlayPhase {
                 for bump_idx in 0..bump {
                     match advancement_policy {
                         // Player *must* defend on Ace and win to advance.
-                        _ if player.rank() == Number::Ace && !(is_defending && bump_idx == 0) => {
+                        _ if landlord.1 != Number::Ace
+                            && player.rank() == Number::Ace
+                            && is_defending =>
```

Comment:
the check as written is incorrect:

1. attacking team can advance past A (!is_defending)
2. if landlord is defending ace, defending team at K can skip past A

Reply:
rule is:

1. all players who did not start on Ace cannot advance past Ace
2. if the player is on the defending team, and if the landlord successfully defended Ace, then the player can advance past Ace

I don't mind if you rewrite the condition here, but the truth table has to be the same

Reply:
> the check as written is incorrect:
> 
> 1. attacking team can advance past A (!is_defending)

> 2. if landlord is defending ace, defending team at K can skip past A

you are right, my bad. Will revert to your check

Reply:
did not realize player's rank was updated inside the loop...